bokeh/document.py

```diff
@@ -356,6 +359,17 @@ def title(self, title):
             self._trigger_on_change(TitleChangedEvent(self, title))
 
     @property
+    def template(self):
+        return self._template
+
+    @template.setter
+    def template(self, template):
+        if not isinstance(template, jinja2.Template):
+            raise ValueError("Document templates must be Jinja2 Templates")
+        self._template = template
+        # TODO (bev) template change eventing?
```

Comment:
I can't think of where change notification would be practically useful right now, fwiw